### PR TITLE
Switch to use GridPartitioner for relationship manager tests

### LINKED ISSUE #14108: Upgrade PETSc submodule to 3.12
## Reason

PETSc 3.12 was released a couple of hours ago.  It contain some new features we are quite interested in. 

## Design

Update PETSc submodule to support this release so that MOOSE users can try these cutting-edge features.

## Impact

Advanced MOOSE users

### PR DESCRIPTION
**Pull request by fdkong**
ParMetis partition is platform dependent. It may also vary for different PETSc releases.

For example, ParMETIS generates a different partition when PETSc-3.12 is used.

For testing relationship managers, we switch to use GridPartitioner that produces the same partition regardless the platform.

Closes #14108

### CODE REVIEW

File: test/tests/relationship_managers/geometric_neighbors/tests [hunk 1]
```diff
@@ -16,10 +17,9 @@
   [./geometric_edge_neighbor_3D]
     type = 'Exodiff'
     input = 'geometric_edge_neighbors.i'
-    cli_args = 'Mesh/dim=3 Mesh/nz=8 Outputs/file_base=geometric_edge_neighbors_3D_out'
+    cli_args = 'Mesh/dim=3 Mesh/nz=8 Outputs/file_base=geometric_edge_neighbors_3D_out Mesh/Partitioner/type=GridPartitioner Mesh/Partitioner/nx=1 Mesh/Partitioner/ny=3'
```
Comment (permcody) [inline test/tests/relationship_managers/geometric_neighbors/tests]: Let's update the input file instead. There's no reason to use the command line override here.

**Review by permcody — CHANGES_REQUESTED** [review]

File: test/tests/relationship_managers/geometric_neighbors/tests (hunk 1 above)
Comment (fdkong) [reply test/tests/relationship_managers/geometric_neighbors/tests]: I do have a reason. GridPartitioner does not adjust the number of subdomains when you try different core counts.  We do not want users to hit errors as follow:

```
kongf$ mpirun -n 4  ../../../moose_test-opt -i geometric_edge_neighbors.i





*** ERROR ***
Number of cells in the GridPartitioner must match the number of MPI ranks! 3 != 4






*** ERROR ***
Number of cells in the GridPartitioner must match the number of MPI ranks! 3 != 4






*** ERROR ***
Number of cells in the GridPartitioner must match the number of MPI ranks! 3 != 4


application called MPI_Abort(MPI_COMM_WORLD, 1) - process 2




*** ERROR ***
Number of cells in the GridPartitioner must match the number of MPI ranks! 3 != 4
```

Comment (lindsayad) [reply test/tests/relationship_managers/geometric_neighbors/tests]: So the point is that you want users to be able to come along and run this with an arbitrary number of processors, independent of the tests spec file?

Comment (permcody) [reply test/tests/relationship_managers/geometric_neighbors/tests]: I'm not sure if Alex's comment is for me or Fande. I'm guess it's for Fande. If it's for me - I was looking at it more like a black box. I'm seeing a bunch of CLI arguments that are the same for both of the two tests that use that input file so I figured, why supply them as CLI arguments versus just putting them in the input file?

For Fande:
This is only a test so I'm not too concerned that people will come and try to run it stand-alone. However, as a separate issue: is there any reason not to auto-update the number cells in the GridPartitioner automatically? With the current design, anytime you put the Grid Partitioner in an input file you are locking it down to a single set of processors. That seems a little heavy handed. Why not automatically adjust the cells? You could print out a `mooseInfo` or `mooseWarning` as appropriate but I don't really see a disadvantage too that.

Comment (fdkong) [reply test/tests/relationship_managers/geometric_neighbors/tests]: I agreed. I think it is a good idea to let GridPartitioner to adjust the number of cells according to the desired  core count.

Comment (lindsayad) [reply test/tests/relationship_managers/geometric_neighbors/tests]: So is that going to get added to this PR? If not, then I think I would actually side with @fdkong on this one. The `min_parallel` and `max_parallel` specifications are in the `tests` file, so it seems consistent to me to specify the `cli_args` corresponding to that parallel specification also in the `tests` file. It does seem weird if someone ever does want to run that input file that we would require them to run with a magnical number of procs. If they didn't look at the `tests` file, they would have no idea how to run it without getting an error.

Comment (permcody) [reply test/tests/relationship_managers/geometric_neighbors/tests]: Yeah - that's reasonable. Of course, we'll forget to change this test later, but whatever.

**Review by permcody — APPROVED** [review]

**Pull request merged**